src/library/baseexternallibraryfeature.cpp

```diff
@@ -125,6 +133,30 @@ void BaseExternalLibraryFeature::slotImportAsMixxxPlaylist() {
     }
 }
 
+void BaseExternalLibraryFeature::slotImportAsMixxxCrate() {
+    // qDebug() << "slotImportAsMixxxCrate() row:" << m_lastRightClickedIndex.data();
+
+    QList<TrackId> trackIds;
+    QString playlist;
+    appendTrackIdsFromRightClickIndex(&trackIds, &playlist);
+    if (trackIds.isEmpty()) {
+        return;
+    }
+
+    Crate crate;
+    crate.setName(playlist);
+
+    CrateId crateId;
+
+    if (m_pTrackCollection->insertCrate(crate, &crateId)) {
+        m_pTrackCollection->addCrateTracks(crateId, trackIds);
+    } else {
+        QMessageBox::warning(nullptr,
+                tr("Crate Creation Failed"),
+                tr("Could not create crate, it most likely already exists: ") + playlist);
```

Comment:
would be nice to have a better error message here but I guess thats not currently possible with the `insertCrate` API, right?